### DISCUSSION TITLE
Historical RT tickets

### DISCUSSION THREAD

**Discussion by davidben** [openssl/openssl#24441] [General Discussion]
Some time ago, OpenSSL maintained an RT instance for its bug tracker. It was eventually [turned down](https://www.openssl.org/blog/blog/2016/10/12/f2f-rt-github/index.html), but a lot of past commits cite RT numbers for their context. For example, https://github.com/openssl/openssl/commit/23f6eec71dbd472044db7dc854599f1de14a1f48 sites "RT#3576" for the key setup, but there is no information in comments or commit message about what this was about, or why there were two.

The old openssl-dev archives contains [a breadcrumb](https://mta.openssl.org/pipermail/openssl-dev/2015-May/001473.html) of context, but it turns out most of that discussion goes back before OpenSSL's openssl-dev archives.

I eventually found some context on [different mailing list archive](https://groups.google.com/g/mailing.openssl.dev/c/OuFXwW4NfO8/m/7d2ZXVjkxVkJ), but this was not easy to find. Did OpenSSL archive its old data somewhere else, or is was the context behind its older decisions purged?

(@levitte, @richsalz suggested that you might know about this?)

**Comment by levitte** [comment]
I don't know much about this one.

But like you're kinda pointing out, commit messages shouldn't just contain references to diverse external systems, they should tell a fuller story of the commit.  We haven't always been the best in that regard.

**Reply by davidben** [reply]
> But like you're kinda pointing out, commit messages shouldn't just contain references to diverse external systems, they should tell a fuller story of the commit.

Those commits indeed could have used more context, but referencing tickets for broader context is pretty natural. At the time RT was OpenSSL's ticket tracker, it's pretty natural that commits would reference it. Given that, it would have been appropriate for the project to maintain _some_ kind of archive of the historical data.

**Comment by briansmith** [comment]
Well, I see now that you found that the discussions are archived on openssl-dev. my understanding is that the openssl-dev web archive scrubs the attachments but if you were subscribed to openssl-dev at the time then you might still have the attachments in your email. So even somebody going into their email program and dumping all the email to a file that gets uploaded somewhere would be useful, so that the attachments can be included in the archive.
